desktop-widgets/tab-widgets/TabDiveInformation.cpp

```diff
@@ -130,10 +147,64 @@ void TabDiveInformation::divesChanged(dive_trip *trip, const QVector<dive *> &di
 	case DiveField::WATER_TEMP:
 		ui->waterTemperatureText->setText(get_temperature_string(current_dive->watertemp, true));
 		break;
+	case DiveField::ATM_PRESS:
+		ui->atmPressVal->setText(ui->atmPressVal->text().sprintf("%d",current_dive->surface_pressure.mbar));
+		break;
 	case DiveField::DATETIME:
 		updateWhen();
 		break;
 	default:
 		break;
 	}
 }
+
+void TabDiveInformation::updateTextBox(int i)
+{
+	pressure_t atmpress = { 0 };
+	double altitudeVal;
+	if (current_dive) {
+		switch (ui->atmPressType->currentIndex())
+		case 0:		// If an atm pressure has been specified in mbar:
+			if (i == 1) {	// this is only triggered by on_atmPressVal_editingFinished()
+				atmpress.mbar = ui->atmPressVal->text().toInt();	// use the specified mbar pressure
+			break;
+		case 1:		// If an altitude has been specified:
+			if (i == 1) {	// this is only triggered by on_atmPressVal_editingFinished()
+				altitudeVal = (ui->atmPressVal->text().toFloat());	// get altitude from text box
+				if (prefs.units.length == units::FEET)		// if altitude in feet
+					altitudeVal = feet_to_mm(altitudeVal); // imperial: convert altitude from feet to mm
+				else
+					altitudeVal = altitudeVal * 1000;	// metric: convert altitude from meters to mm
+				atmpress.mbar = altitude_to_pressure((int32_t) altitudeVal); // convert altitude (mm) to pressure (mbar)
+				ui->atmPressVal->setText(ui->atmPressVal->text().sprintf("%d",atmpress.mbar));
+				ui->atmPressType->setCurrentIndex(0);		// reset combobox to mbar
+			}
+			if (i == 0)
```

Comment:
This could also be written as `else...` or `else if...`

Reply:
Done